src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -380,6 +382,9 @@ protected void ExecuteIndexing()
 
                             DoIndexingWork(stats, cts.Token);
 
+                            _indexingBatchCompleted.SetByAsyncCompletion();
+                            _indexingBatchCompleted.Reset();
+
```

Comment:
I wonder if we are better off with auto reset event here?
http://blogs.msdn.com/b/pfxteam/archive/2012/02/11/10266923.aspx